serverless.yml

```
@@ -9,22 +9,69 @@ provider:
       Action:
         - 's3:PutObject'
         - 's3:GetObject'
-        - 's3:PutObjectAcl'
         - 's3:GetObjectTagging'
         - 's3:PutObjectTagging'
+        - 's3:GetObjectAcl'
+        - 's3:PutObjectAcl'
+        - 's3:GetBucketAcl'
+        - 's3:PutBucketAcl'
       Resource: 'arn:aws:s3:::${self:custom.bucket_name}/*'
     - Effect: Allow
       Action:
         - 'lambda:InvokeFunction'
         - 'lambda:InvokeAsync'
       Resource: '*'
+    # allow to attach events to exiting buckets (serverless-plugin-existing-s3 plugin)
+    -  Effect: "Allow"
+       Action:
+         - "s3:PutBucketNotification"
+       Resource:
+         Fn::Join:
+           - ""
+           - - "arn:aws:s3:::${self:provider.environment.ORIGIN_BUCKET}"
```

this seems to be a typo
```suggestion
           - "arn:aws:s3:::${self:provider.environment.ORIGIN_BUCKET}"
```

---

not a typo

---

why the double dashes then?

---

This is how it looks if you express it in json
```
{ Fn::Join: [
	"",
	["arn:aws:s3:::${self:provider.environment.ORIGIN_BUCKET}"]
  ]
}      
```